asset-transfer-basic/application-gateway-go/assetTransfer.go

```diff
@@ -0,0 +1,258 @@
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+	"time"
+	"strconv"
+	"github.com/hyperledger/fabric-gateway/pkg/client"
+	"github.com/hyperledger/fabric-gateway/pkg/identity"
+	gwproto "github.com/hyperledger/fabric-protos-go/gateway"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+const (
+
+	mspID        = "Org1MSP"
+	cryptoPath   = "../../test-network/organizations/peerOrganizations/org1.example.com"
+	certPath     = cryptoPath + "/users/User1@org1.example.com/msp/signcerts/cert.pem"
+	keyPath      = cryptoPath + "/users/User1@org1.example.com/msp/keystore/"
+	tlsCertPath  = cryptoPath + "/peers/peer0.org1.example.com/tls/ca.crt"
+	peerEndpoint = "localhost:7051"
+	gatewayPeer  = "peer0.org1.example.com"
+	channelName  = "mychannel"
+	chaincodeName= "basic"
+)
+	var assetId  = "asset" + strconv.FormatInt(time.Now().Unix(),10);
+
+func main() {
+	log.Println("============ application-golang starts ============")
+
+	// The gRPC client connection should be shared by all Gateway connections to this endpoint
+	clientConnection := newGrpcConnection()
+	defer clientConnection.Close()
+
+	id := newIdentity()
+	sign := newSign()
+
+	// Create a Gateway connection for a specific client identity
+	gateway, err := client.Connect(
+		id,
+		client.WithSign(sign),
+		client.WithClientConnection(clientConnection),
+		// Default timeouts for different gRPC calls
+		client.WithEvaluateTimeout(5*time.Second),
+		client.WithEndorseTimeout(15*time.Second),
+		client.WithSubmitTimeout(5*time.Second),
+		client.WithCommitStatusTimeout(1*time.Minute),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer gateway.Close()
+
+	network := gateway.GetNetwork(channelName)
+	contract := network.GetContract(chaincodeName)
+
+	fmt.Println("initLedger:")
+	initLedger(contract);
+
+	fmt.Println("getAllAssets:")
+	getAllAssets(contract);
+
+	fmt.Println("createAsset:")
+	createAsset(contract);
+
+	fmt.Println("readAssetByID:")
+	readAssetByID(contract);
+
+	fmt.Println("transferAssetAsync:")
+    transferAssetAsync(contract)
+
+	fmt.Println("exampleErrorHandling:")
+	exampleErrorHandling(contract)
+
+	log.Println("============ application-golang ends ============")
+}
+
+
+
+func initLedger(contract *client.Contract){
+	fmt.Printf("Submitting \"InitLedger\" transaction. \n", )
+	// Submit transaction, blocking until the transaction has been committed on the ledger
+	_, err := contract.SubmitTransaction("InitLedger")
+	if err != nil {
+		panic(fmt.Errorf("failed to submit transaction: %w", err))
+	}
+	fmt.Printf("InitLedger Transaction successful. \n")
+
+
+}
+
+
+func getAllAssets(contract *client.Contract){
+	fmt.Println("Evaluating \"GetAllAssets\" query.")
+	evaluateResult, err := contract.EvaluateTransaction("GetAllAssets")
+	if err != nil {
+		panic(fmt.Errorf("failed to evaluate transaction: %w", err))
+	}
+	fmt.Printf("GetAllAssets query result = %s\n", string(evaluateResult))
```

Comment:
Check the output from this line when the sample is run. I suspect the JSON string will be collapsed to a single line with no formatting and very hard to read. If so, it would be worth trying to format it, perhaps with [json.Indent()](https://pkg.go.dev/encoding/json#Indent)